multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
+        Opens button to create a JSON of the cell locations
+        """
+        self.base_dir = self.mosaic_editor.ui.fileLoader.baseDir()
+        if self.base_dir.info()['dirType'] == 'Slice':
+            self.base_path = self.base_dir.name()
+            self.slice_name = self.base_dir.info()['specimen_ID']
+            self.slice_id = lims.specimen_id_from_name(self.slice_name)
+            clusters = lims.cell_cluster_ids(self.slice_id)
+            """if len(clusters) == 0:
+                                                    raise Exception("No Clusters Recorded in LIMS")"""
+            try:
+                aff_image_path = lims.specimen_20x_image(self.slice_name)
+                #check the image file path
+                safe_aff_image_path = lims.lims.safe_system_path(aff_image_path)
+
+                if os.path.exists(safe_aff_image_path) == True:
+                    print('File Exists in LIMS')
+                    aff_image_name = aff_image_path.split("/")[-1]
+                    jpg_image_name = aff_image_name.split(".aff")[0] + ".jpg"
+
+                    image_service_url = "http://lims2/cgi-bin/imageservice?path="
+                    zoom = 5        #need to check that this ACQ4 can support this size
+
+                    full_url = image_service_url + aff_image_path + "&zoom={}".format(str(zoom))
+                    print(full_url)
+                    save_path = self.base_path + '/' + jpg_image_name
+                    safe_save_path = lims.lims.safe_system_path(save_path)
+
+                    if os.path.exists(safe_save_path) == True:
+                        print("20x already moved")
+
+                    elif os.path.exists(safe_save_path) == False:
+                        image = urllib.URLopener()
+                        image.retrieve(full_url, safe_save_path)
+                        print("20x image moved")
+                    self.image_20 = safe_save_path
+                        
+                else:
+                    raise Exception("Couldn't find image in LIMS. Check you have the selected the correct slice.")
+                
+            except KeyError:
+                raise Exception('No Slice Selected')
+        else:
+            raise Exception('No Slice Selected')
+       
+
+
+    def create_lims_json(self):
+        """
+        Checks that markers are loaded into the canvas
+        Gets timestamp
+        Creates json for each cell clusters and saves locally
+        copies json to correct incoming folder
+        creates trigger file and drops in correct incoming folder
+        """
+        
+        items = self.mosaic_editor.canvas.items 
+        markers = [i for i in items if isinstance(i, MarkersCanvasItem)]
+        if len(markers) != 1:
+            raise Exception("Must have exactly 1 Markers item in the canvas.")
+        
+        for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
```

Comment:
Make sure there is only one item that matches this description. OR: load the image in `load_clicked` and store a reference to the item that was created.

Reply:
see above about loading 20x after moving from LIMS